common/src/math.rs

```diff
@@ -3,97 +3,463 @@
+        MIsometry(self.0.map(f))
+    }
+    #[inline]
+    pub fn from_columns_unchecked(columns: &[MVector<N>;4]) -> Self
+    {
+        Self(na::Matrix4::from_columns(&(*columns).map(|x| x.0)))
+    }
+    /// Minkowski transpose. Inverse for hyperbolic isometries
+    #[rustfmt::skip]
+    pub fn mtranspose(self) -> Self {
+        MIsometry(
+            na::Matrix4::new(
+                self.0.m11,  self.0.m21,  self.0.m31, -self.0.m41,
+                self.0.m12,  self.0.m22,  self.0.m32, -self.0.m42,
+                self.0.m13,  self.0.m23,  self.0.m33, -self.0.m43,
+                -self.0.m14, -self.0.m24, -self.0.m34,  self.0.m44,
+            )
+        )
+    }
+    /// Whether an isometry reverses winding with respect to the norm
+    pub fn parity(self) -> bool {
+        self.0.fixed_view::<3, 3>(0, 0).determinant() < na::zero::<N>()
+    }
+    pub fn renormalize_isometry(self) -> MIsometry<N>
+    {
+        let boost = translate(&MVector::origin(), &MVector(self.0.index((.., 3)).into()).lorentz_normalize());
+        let inverse_boost = boost.mtranspose();
+        let rotation = renormalize_rotation_reflection(
+            &((inverse_boost * self).0).fixed_view::<3, 3>(0, 0).clone_owned(),
+        );
+        MIsometry(boost.0 * rotation.to_homogeneous())
+    }
+    /*pub fn cast<N2: Scalar>(self) -> MIsometry<N2> where N: simba::scalar::SubsetOf<N2>
+    {
+        Self(self.0.cast::<N2>())
+    }*/
+}
+
+impl<N: RealField + Copy> MVector<N> {
+    pub fn lorentz_normalize(self: &MVector<N>) -> Self {
+        let sf2 = self.mip(self);
+        if sf2 == na::zero() {
+            return MVector::origin();
+        }
+        let sf = sf2.abs().sqrt();
+        *self / sf
+    }
+    /// Point or plane reflection around point or normal `p`
+    pub fn reflect(self) -> MIsometry<N> {
+        MIsometry(na::Matrix4::<N>::identity()
+            - self.minkowski_outer_product(&self) * na::convert::<_, N>(2.0) / self.mip(&self))
+    }
+    /// Minkowski inner product, aka <a, b>_h
+    pub fn mip(self, other: &Self) -> N
+    {
+        self.0.x * other.0.x + self.0.y * other.0.y + self.0.z * other.0.z - self.0.w * other.0.w
+    }
+    pub fn minkowski_outer_product(self, other: &Self) -> na::Matrix4<N>
+    {
+        ((self).0) * na::RowVector4::new(other.0.x, other.0.y, other.0.z, -other.0.w)
+    }
+    pub fn from_gans(gans: &na::Vector3<N>) -> Self
+    {
         // x^2 + y^2 + z^2 - w^2 = -1
         // sqrt(x^2 + y^2 + z^2 + 1) = w
-        let w = (sqr(self.0.x) + sqr(self.0.y) + sqr(self.0.z) + na::one()).sqrt();
-        na::Vector4::new(self.0.x, self.0.y, self.0.z, w)
+        let w = (sqr(gans.x) + sqr(gans.y) + sqr(gans.z) + na::one()).sqrt();
+        MVector(na::Vector4::new(gans.x, gans.y, gans.z, w))
+    }
+    #[inline]
+    pub fn zero() -> Self
+    {
+        Self(na::zero())
+    }
+    pub fn origin() -> Self
+    {
+        Self(na::Vector4::new(na::zero(),na::zero(),na::zero(),na::one()))
```

Comment:
I would recommend simplifying this to
```rust
pub fn origin() -> Self {
    Self::w()
}
```

Personally, I'm not sure if we should have the `origin()` function at all, since it's equivalent to `w()`, but that was already true before this PR, so in the spirit of keeping changes incremental, I would recommend keeping `origin()` around for now, but we might as well make its equivalence to `w()` clearer.

Reply:
id be fine with changing the definition to ``w()``, but having ``w()`` be solely used feels a bit like obscuring the meaning of it

Reply:
The suggestion is to change the definition, not remove it. I think we all agree that `origin()` is hugely valuable for readability.